src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -1027,69 +1025,321 @@ private static void ParseStatusLine(ReadOnlySpan<byte> line, HttpResponseMessage
+
+                            vector = Avx.LoadAlignedVector256(vectorIter);
+                            foundLF = (uint)Avx2.MoveMask(Avx2.CompareEqual(vector, maskLF));
+                            foundCol = foundLF | (uint)Avx2.MoveMask(Avx2.CompareEqual(vector, maskCol));
+                        }
 
-            if (!HeaderDescriptor.TryGet(line.Slice(0, pos), out HeaderDescriptor descriptor))
+                        foundCol ^= 1u << colIdx;
+                        col = vectorIter + colIdx;
+                    }
+                    // Branch will not be taken unless padding bytes prior to vectorBegin match.
+                    while (col < nameStart);
+
+                    if (col >= vectorEnd)
+                    {
+                        // Branch will not be taken unless padding bytes beyond vectorEnd match.
+                        goto needMore;
+                    }
+
+                    if (*col == '\n')
+                    {
+                        if ((col == nameStart + 1 && *nameStart == '\r') || col == nameStart)
+                        {
+                            // End of headers.
+                            return (finished: true, bytesConsumed: (int)(col - vectorBegin + 1));
+                        }
+                        else
+                        {
+                            // Found a newline without a colon.
+                            buffer = new Span<byte>(nameStart, (int)(col - nameStart));
+                            throw new HttpRequestException(SR.Format(SR.net_http_invalid_response_header_line, Encoding.ASCII.GetString(buffer)));
+                        }
+                    }
+
+                    // Found the start of header value.
+                    byte* valueStart = col + 1;
+                    if (valueStart == vectorEnd)
+                    {
+                        goto needMore;
+                    }
+
+                    // Find the end of the value, a CRLF or LF.
+                    byte* crOrLf;
+                    byte* lf;
+                    while (true)
+                    {
+                        do
+                        {
+                            int lfIdx;
+                            while ((lfIdx = BitOperations.TrailingZeroCount(foundLF)) == Vector256<byte>.Count)
+                            {
+                                vectorIter += Vector256<byte>.Count;
+                                if (vectorIter >= vectorEnd)
+                                {
+                                    goto needMore;
+                                }
+
+                                vector = Avx.LoadAlignedVector256(vectorIter);
+                                foundLF = (uint)Avx2.MoveMask(Avx2.CompareEqual(vector, maskLF));
+                                foundCol = foundLF | (uint)Avx2.MoveMask(Avx2.CompareEqual(vector, maskCol));
```

Comment:
It seems like this is unnecessary work in a fair amount of cases... i.e. if the header value extends beyond this vector, which seems pretty common. Would it make sense to defer this until we actually find the \n indicating end of header value?